src/confluent_kafka/schema_registry/json_schema.py

```diff
@@ -238,7 +277,15 @@ def __call__(self, obj, ctx):
             value = obj
 
         try:
-            validate(instance=value, schema=self._parsed_schema)
+            if self._are_references_provided:
+                named_schemas = _resolve_named_schema(self._schema, self._registry)
+                if len(named_schemas) > 1:
```

Comment:
Remove this check as we are already checking this earlier while assigning `self._are_references_provided`